targets/TARGET_Ambiq_Micro/TARGET_Apollo3/device/i2c_api.c

```diff
@@ -0,0 +1,251 @@
+/*
+Copyright (c) 2020 SparkFun Electronics
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+#if DEVICE_I2C
+
+#include "i2c_api.h"
+#include "iom_api.h"
+#include "PeripheralPins.h"
+#include "mbed_assert.h"
+
+#define DEFAULT_CLK_FREQ (AM_HAL_IOM_400KHZ)
+
+#define standin_fn() printf("stand in for '%s', file: '%s', line: %d\n", __PRETTY_FUNCTION__, __FILE__, __LINE__)
+
+static am_hal_iom_transfer_t xfer = {0};
+
+I2CName i2c_get_peripheral_name(PinName sda, PinName scl){
+    uint32_t iom_sda = pinmap_peripheral(sda, i2c_master_sda_pinmap());
+    uint32_t iom_scl = pinmap_peripheral(scl, i2c_master_scl_pinmap());
+
+    uint32_t iom = pinmap_merge(iom_sda, iom_scl);
+
+    if((int)iom == NC){
+        return IOM_NUM;
+    }
+
+    return (I2CName)iom;
+}
+
+void i2c_init(i2c_t *obj, PinName sda, PinName scl){
+    MBED_ASSERT(obj);
+
+    // iom determination
+    I2CName iom = i2c_get_peripheral_name(sda, scl);
+    MBED_ASSERT((int)iom != IOM_NUM);
+
+    // iom configuration
+    obj->i2c.iom_obj.iom.inst = (uint32_t)iom;
+    obj->i2c.iom_obj.iom.cfg.eInterfaceMode = AM_HAL_IOM_I2C_MODE;
+    obj->i2c.iom_obj.iom.cfg.ui32ClockFreq = DEFAULT_CLK_FREQ;
+    obj->i2c.iom_obj.iom.cfg.pNBTxnBuf = NULL;
+    obj->i2c.iom_obj.iom.cfg.ui32NBTxnBufLength = 0;
+
+    // pin configuration
+    if((int)sda != NC){ pinmap_config(sda, i2c_master_sda_pinmap()); }
```

Comment:
Can you run astyle on these files in `TARGET_Apollo3/device` ? There are HAL implementation and should follow Mbed OS coding style (using configuration .astylerc in Mbed OS root)

Reply:
done